include/zephyr/mgmt/hawkbit.h

```diff
@@ -94,12 +95,17 @@ void hawkbit_autohandler(void);
 /**
  * @brief The hawkBit probe verify if there is some update to be performed.
  *
- * @return HAWKBIT_UPDATE_INSTALLED has an update available.
- * @return HAWKBIT_NO_UPDATE no update available.
- * @return HAWKBIT_NETWORKING_ERROR fail to connect to the hawkBit server.
- * @return HAWKBIT_METADATA_ERROR fail to parse or to encode the metadata.
- * @return HAWKBIT_OK if success.
- * @return HAWKBIT_DOWNLOAD_ERROR fail while downloading the update package.
+ * @retval HAWKBIT_NETWORKING_ERROR fail to connect to the hawkBit server.
+ * @retval HAWKBIT_UNCONFIRMED_IMAGE image is unconfirmed.
+ * @retval HAWKBIT_PERMISSION_ERROR fail to get the permission to access the hawkBit server.
+ * @retval HAWKBIT_METADATA_ERROR fail to parse or to encode the metadata.
+ * @retval HAWKBIT_DOWNLOAD_ERROR fail while downloading the update package.
+ * @retval HAWKBIT_OK if the image was already updated.
+ * @retval HAWKBIT_UPDATE_INSTALLED if an update was installed. Reboot is required to apply it.
+ * @retval HAWKBIT_NO_UPDATE if no update was available.
+ * @retval HAWKBIT_CANCEL_UPDATE if the update was cancelled by the server.
+ * @retval HAWKBIT_NOT_INITIALIZED if hawkBit is not initialized.
+ * @retval HAWKBIT_PROBE_IN_PROGRESS if probe is currently running.
```

Comment:
not really blocking but it feels like this could be ordered better, ex. having  `HAWKBIT_OK` listed first?

Reply:
it's ordered in the same way as the `enum hawkbit_response`. 
`HAWKBIT_OK` is actually not the standard return value, if everything is successful. Its name is a bit misleading. It is only returned one time after installing an update, rebooting, and then informing the server about the successful update.  

BTW `HAWKBIT_OK` will be removed with https://github.com/zephyrproject-rtos/zephyr/pull/70787. After that we will only have `HAWKBIT_UPDATE_INSTALLED`, `HAWKBIT_NO_UPDATE` and the errors as return value. (And it will be ordered.)